### PR TITLE
105 scholars discovery okd setup

### PR DESCRIPTION
**Pull request by ht29**

### CODE REVIEW

File: solr/Dockerfile.okd [hunk 1]
```diff
@@ -0,0 +1,59 @@
+FROM solr:7.7.2-slim
+
+#COPY --chown=solr:solr config /opt/solr/server/solr/persons
+#COPY --chown=solr:solr config /opt/solr/server/solr/organizations
+#COPY --chown=solr:solr config /opt/solr/server/solr/documents
+#COPY --chown=solr:solr config /opt/solr/server/solr/relationships
+#COPY --chown=solr:solr config /opt/solr/server/solr/processes
+#COPY --chown=solr:solr config /opt/solr/server/solr/collections
+#COPY --chown=solr:solr config /opt/solr/server/solr/concepts
```
Comment (outten45) [inline solr/Dockerfile.okd]: hi @ht29 should we removed the commented code? we can always compare with the `Dockerfile` in the directory.

File: solr/Dockerfile.okd [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+RUN chmod ugo+w -R /opt/solr/server/solr/
+RUN chmod ugo+w -R /opt/solr/server/logs/
+
+USER solr
+
+RUN mkdir -p /opt/solr/server/solr/persons
```
Comment (outten45) [inline solr/Dockerfile.okd]: hi @ht29 maybe we should add a commented section here explaining a little why we had to separate these calls. something like:

```
# calls had to be separated into multiple commands
# in order to support OpenShift environment.
```

**Comment by wwelling** [comment]
What is it about OpenShift that does not allow this to work? I notice the `COPY` command changed its source to `./solr/config`. Does this work if running `docker build` from the directory of this Dockerfile?

**Comment by outten45** [comment]
hi @wwelling I believe docker within our openshift environment doesn't support the `--chown=solr:solr ` option within the COPY command. @ht29 that's correct, right? the instance of OKD we are running might be slightly older since it doesn't support multi-stage builds either. we didn't want to replace the current one since hopefully we can switch to the cleaner version in the future.

**Comment by wwelling** [comment]
If that’s the case, the multiple commands for copy would be fine. I am a bit concerned about the source of the copy changing. I will test locally and see if the `COPY` source can be `./solr/config`.

File: solr/Dockerfile.okd (hunk 1 above)
Comment (ht29) [reply solr/Dockerfile.okd]: yes.. @outten45 .. I can remove the commented code...thanks

File: solr/Dockerfile.okd (hunk 2 above)
Comment (ht29) [reply solr/Dockerfile.okd]: Hi @outten45 , I think it is a good idea.. I can add some meaningful comments in there for explaination...thanks

**Comment by wwelling** [comment]
I have no way of running on OpenShift, but would something like this work? Maybe recursively change ownership of `/opt/solr/server/solr` after copying configurations as well.

```
FROM solr:7.7.2-slim

RUN chown -R solr /opt/solr/server/solr

RUN chmod ugo+w -R /opt/solr/server/solr/
RUN chmod ugo+w -R /opt/solr/server/logs/

USER solr 

COPY ./config /opt/solr/server/solr/persons
COPY ./config /opt/solr/server/solr/organizations
COPY ./config /opt/solr/server/solr/documents
COPY ./config /opt/solr/server/solr/relationships
COPY ./config /opt/solr/server/solr/processes
COPY ./config /opt/solr/server/solr/collections
COPY ./config /opt/solr/server/solr/concepts

RUN precreate-core persons
RUN precreate-core organizations
RUN precreate-core documents
RUN precreate-core relationships
RUN precreate-core processes
RUN precreate-core collections
RUN precreate-core concepts
```

**Comment by wwelling** [comment]
What is common practice for where docker build is run from? I understand the case where running it from the root of the repo. I feel as if docker build is run from anywhere else the copy source will have to be changed in the Dockerfile. I deferred to running docker build from the directory of the Dockerfile. If one chooses to run docker build from another directory would have to have the copy source relative to their current working directory. In this case would be able to have various config in different locations, in which could build images with different config using the same Dockerfile.

**Comment by wwelling** [comment]
I am not apposed to setting expectation of running docker build from the root of the repo and explicitly providing the Dockerfile.

**Comment by ht29** [comment]
Hi @wwelling  and @outten45 ,

I was able to fix the config location issue in `Dockerfile.okd` and now it will successfully build locally from `solr` folder and from the openshift enviornment as well. Please, look at the latest commit and hopefully that answers your questions. Sorry for repetitive commit messages..

**Comment by wwelling** [comment]
https://github.com/vivo-community/scholars-discovery/blob/master/solr/config/managed-schema

https://scholars-discovery-solr-scholars.cloud.duke.edu/solr/#/collections/files?file=managed-schema

Need to pass the config location to `precreate-core`. It is currently just using default configurations.

**Comment by ht29** [comment]
@wellingWilliam , please check it out now.. should be good..

**Comment by wwelling** [comment]
https://scholars-discovery-solr-scholars.cloud.duke.edu/solr/#/collections/files now fails

**Comment by wwelling** [comment]
Does appear to have the correct schema fields now.

**Comment by ht29** [comment]
Hi @wwelling .. yes, the schema does have correct fields and everything looks good. It just can't list Files under any of the cores... I am looking at it but I think it should not stop any work for now..

**Comment by ht29** [comment]
Hi @wellingWilliam , I think it is fixed now. Please check it out.

**Comment by ht29** [comment]
Hi @wwelling , to make it completely simple, I had to change permission on config folder by doing `chmod 777 config` itself in the repo to make it writable before it copies in the container. Hence, in the pull request you will see lot of files but really it is just permission change which is right way to do anyways.

**Comment by wwelling** [comment]
That’s great. I don’t think we need two docker files now. This is even cleaner than the previous.

**Comment by wwelling** [comment]
Unfortunately, the cores used default config.

https://scholars-discovery-solr-scholars.cloud.duke.edu/solr/#/collections/files?file=solrconfig.xml

**Comment by ht29** [comment]
Hi @wwelling , it should be working now..

**Comment by wwelling** [comment]
@ht29 testing locally now

**Comment by wwelling** [comment]
Starts up good. The throws exception when `scholars-discovery` connects to manage the schema.

```
2019-08-21 14:27:23.277 ERROR (qtp1107530534-16) [   x:organizations] o.a.s.s.ManagedIndexSchema Error persisting managed schema /opt/solr/server/solr/mycores/organizations/conf/managed-schema
java.io.FileNotFoundException: /opt/solr/server/solr/mycores/organizations/conf/managed-schema (Permission denied)
        at java.io.FileOutputStream.open0(Native Method) ~[?:?]
        at java.io.FileOutputStream.open(FileOutputStream.java:298) ~[?:?]
        at java.io.FileOutputStream.<init>(FileOutputStream.java:237) ~[?:?]
        at java.io.FileOutputStream.<init>(FileOutputStream.java:187) ~[?:?]
        at org.apache.solr.schema.ManagedIndexSchema.persistManagedSchema(ManagedIndexSchema.java:133) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.schema.SchemaManager.doOperations(SchemaManager.java:139) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.schema.SchemaManager.performOperations(SchemaManager.java:83) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.handler.SchemaHandler.handleRequestBody(SchemaHandler.java:90) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:199) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.core.SolrCore.execute(SolrCore.java:2551) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.servlet.HttpSolrCall.execute(HttpSolrCall.java:711) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.servlet.HttpSolrCall.call(HttpSolrCall.java:516) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:395) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:341) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1602) [jetty-servlet-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:540) [jetty-servlet-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548) [jetty-security-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1588) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480) [jetty-servlet-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1557) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:335) [jetty-rewrite-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.Server.handle(Server.java:502) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305) [jetty-io-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [jetty-io-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118) [jetty-io-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at java.lang.Thread.run(Thread.java:834) [?:?]
2019-08-21 14:27:23.308 ERROR (qtp1107530534-16) [   x:organizations] o.a.s.s.ManagedIndexSchema Error syncing the managed schema file /opt/solr/server/solr/mycores/organizations/conf/managed-schema
java.io.FileNotFoundException: /opt/solr/server/solr/mycores/organizations/conf/managed-schema (Permission denied)
        at java.io.RandomAccessFile.open0(Native Method) ~[?:?]
        at java.io.RandomAccessFile.open(RandomAccessFile.java:345) ~[?:?]
        at java.io.RandomAccessFile.<init>(RandomAccessFile.java:259) ~[?:?]
        at java.io.RandomAccessFile.<init>(RandomAccessFile.java:214) ~[?:?]
        at org.apache.solr.util.FileUtils.sync(FileUtils.java:77) ~[solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.schema.ManagedIndexSchema.persistManagedSchema(ManagedIndexSchema.java:144) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.schema.SchemaManager.doOperations(SchemaManager.java:139) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.schema.SchemaManager.performOperations(SchemaManager.java:83) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.handler.SchemaHandler.handleRequestBody(SchemaHandler.java:90) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:199) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.core.SolrCore.execute(SolrCore.java:2551) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.servlet.HttpSolrCall.execute(HttpSolrCall.java:711) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.servlet.HttpSolrCall.call(HttpSolrCall.java:516) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:395) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:341) [solr-core-7.7.2.jar:7.7.2 d4c30fc2856154f2c1fefc589eb7cd070a415b94 - janhoy - 2019-05-28 23:37:48]
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1602) [jetty-servlet-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:540) [jetty-servlet-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548) [jetty-security-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1588) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480) [jetty-servlet-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1557) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:335) [jetty-rewrite-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.Server.handle(Server.java:502) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260) [jetty-server-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305) [jetty-io-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [jetty-io-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118) [jetty-io-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683) [jetty-util-9.4.14.v20181114.jar:9.4.14.v20181114]
        at java.lang.Thread.run(Thread.java:834) [?:?]
2019-08-21 14:27:23.309 WARN  (qtp1107530534-16) [   x:organizations] o.a.s.s.SchemaManager Unable to persist managed schema.
2019-08-21 14:27:23.310 ERROR (qtp1107530534-16) [   x:organizations] o.a.s.h.RequestHandlerBase org.apache.solr.api.ApiBag$ExceptionWithErrObject: error processing commands, errors: [Unable to persist managed schema. Error persisting managed schema /opt/solr/server/solr/mycores/organizations/conf/managed-schema],
        at org.apache.solr.handler.SchemaHandler.handleRequestBody(SchemaHandler.java:92)
        at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:199)
        at org.apache.solr.core.SolrCore.execute(SolrCore.java:2551)
        at org.apache.solr.servlet.HttpSolrCall.execute(HttpSolrCall.java:711)
        at org.apache.solr.servlet.HttpSolrCall.call(HttpSolrCall.java:516)
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:395)
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:341)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1602)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:540)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:146)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:548)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:257)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1588)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:255)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1345)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:203)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:480)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1557)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:201)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1247)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:144)
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:220)
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:126)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:335)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.Server.handle(Server.java:502)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:364)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:260)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:305)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:118)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:333)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:310)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:168)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:126)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:366)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:765)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:683)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by wwelling** [comment]
I have updated `solr/Dockerfile` on a `tamu-sprint9-staging` to match closer.

https://github.com/vivo-community/scholars-discovery/blob/tamu-sprint9-staging/solr/Dockerfile

This works as expected with `scholars-discovery` indexing. I have resolved the issue to the ownership of the configurations. If I have time I will try to get permissions set correctly without the `--chown` attribute of `COPY`.

**Comment by wwelling** [comment]
This works:

```
FROM solr:7.7.2-slim

RUN precreate-core persons
RUN precreate-core organizations
RUN precreate-core documents
RUN precreate-core relationships
RUN precreate-core processes
RUN precreate-core collections
RUN precreate-core concepts

RUN chmod -R ugo+w /opt/solr/server/solr/
RUN chmod -R ugo+w /opt/solr/server/logs/

COPY config /opt/solr/server/solr/mycores/persons/conf
COPY config /opt/solr/server/solr/mycores/organizations/conf
COPY config /opt/solr/server/solr/mycores/documents/conf
COPY config /opt/solr/server/solr/mycores/relationships/conf
COPY config /opt/solr/server/solr/mycores/processes/conf
COPY config /opt/solr/server/solr/mycores/collections/conf
COPY config /opt/solr/server/solr/mycores/concepts/conf

USER root

RUN chown -R solr:solr /opt/solr/server/solr/mycores/

USER solr
```

**Comment by ht29** [comment]
Hi @wwelling and @outten45, your changes works in Openshift as well. Can you test out one more time ? Once confirmed, I will make one version of Dockerfile and update this request so one of you can merge it. Thanks...

**Comment by wwelling** [comment]
Works great. Do we need the `RUN chmod ...` lines? It works locally without them. Can you test in OpenShift without those 2 lines? If it works, I see no reason to keep them in there.

https://github.com/vivo-community/scholars-discovery/pull/140/files#diff-9ebf2223a6c12355a91ecb10502cecfbR11
https://github.com/vivo-community/scholars-discovery/pull/140/files#diff-9ebf2223a6c12355a91ecb10502cecfbR12

**Comment by ht29** [comment]
Hi @wellingWilliam, the permissions are necessary to run in openshift. I just tested them to make sure. Thanks.

**Comment by ht29** [comment]
Hi @wwelling, I made only Dockerfile now which covers for both local and openshift solr container. Can you merge this change in ?? Thanks..

**Comment by wwelling** [comment]
Are you okay with a squash merge? Not much information within all the commits.

**Pull request merged**